### PR TITLE
[review] Add comments on suspected vulnerabilities

### PR DESCRIPTION
**Pull request by SHA-2048**
I've added comments on parts of the code that I think needs to be secured/simplified

most critical issues:
- Use safeTransfer
- Check for overflows when calculating claim amounts

### CODE REVIEW

File: contracts/HundredVesting.sol [hunk 1]
```diff
@@ -6,9 +6,16 @@ import "@openzeppelin/contracts/access/Ownable.sol";
 import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 
+// no Ownable modifiers are used
+// does it really need to inherit
+// from Ownable ?
```
Comment (defi-bear) [inline contracts/HundredVesting.sol:11]: It's correct. It's not needed. I will delete it.

File: contracts/HundredVesting.sol [hunk 2]
```diff
@@ -6,9 +6,16 @@ import "@openzeppelin/contracts/access/Ownable.sol";
...
 contract HundredVesting is Ownable {
     IERC20 public immutable Hundred;
     uint256 totalPeriod = 365 * 24 * 60 * 60;
+    // timePerPeriod seems to be used once
+    // in a formulae where its value is canceled
+    // after expanding the expression
+    // which indicate it is not needed
```
Comment (defi-bear) [inline contracts/HundredVesting.sol:18]: I believe it's needed. Without it, they can claim tokens by sec.
But with it, we can set the period like one day. Then they can get the claimable tokens every day.

File: contracts/HundredVesting.sol [hunk 3]
```diff
@@ -25,10 +32,14 @@ contract HundredVesting is Ownable {
 
     function getRemainedAmount (address vester) internal view returns (uint256) {
         uint256 timestamp = addresses[vester].timestamp;
+        // should probably have a require here to avoid negative values
+        // or use a int256
         uint256 amount = (block.timestamp - timestamp) / timePerPeriod * addresses[vester].amount / totalPeriod;
+        // check return value is always > 0, this can lead to user getting
+        // more tokens after the end of the vesting period
```
Comment (defi-bear) [inline contracts/HundredVesting.sol:39]: Right, I will return amount < 0 ? 0 : amount

File: contracts/HundredVesting.sol [hunk 4]
```diff
@@ -42,20 +53,30 @@ contract HundredVesting is Ownable {
         } else {
             user = UserInfo({ amount: amount, timestamp: block.timestamp, claimedAmount: 0 });
         }
+        // redundant assignment
         user.timestamp = block.timestamp;
```
Comment (defi-bear) [inline contracts/HundredVesting.sol:57]: Yes, I think I forgot to remove it.

File: contracts/HundredVesting.sol [hunk 5]
```diff
@@ -42,20 +53,30 @@ contract HundredVesting is Ownable {
...
         addresses[vester] = user;
     }
 
+    // why vester is a param, shouldn't it be msg.sender ?
+    // also vester is a misleading name, should
+    // probably be more vestedAddress or similar
```
Comment (defi-bear) [inline contracts/HundredVesting.sol:63]: Good point

File: contracts/HundredVesting.sol [hunk 6]
```diff
@@ -42,20 +53,30 @@ contract HundredVesting is Ownable {
...
+        // but either use safeTransfer or increment
+        // claimedAmount after checking transfer
+        // is successful
         addresses[vester].claimedAmount = addresses[vester].claimedAmount + amount;
         Hundred.transfer(msg.sender, amount);
     }
 
     function getClaimableAmount (address vester) public view returns(uint) {
+        // do we really need to check the 0 address ?
+        // if so, can be factored into a modifier maybe...
```
Comment (defi-bear) [inline contracts/HundredVesting.sol:79]: I think, it's not and also parameter should be changed to msg.sender

**Comment by defi-bear** [comment]
@SHA-2048 Thank you for reviewing them.
I just made a new commit but forgot to make a pr. Anyway, could you review the new changes and left a comment here?

**Comment by defi-bear** [comment]
@SHA-2048 I've made pr with the previous commit.
https://github.com/hundred-finance/hundred-migration/pull/2
Could you review it? I tried to request a review from you but I cannot find you on the reviewers' list.

**Comment by vfat0** [comment]
closed via #2

**Pull request closed without merging**